### ISSUE TITLE
vspkg install Shark

### ISSUE THREAD

**Issue by sursu** [Shark-ML/Shark#228]
Not really an issue, but rather a suggestion which I will formulate as a wish:

I wish Shark could be installed via [vcpkg](https://blogs.msdn.microsoft.com/vcblog/2018/04/24/announcing-a-single-c-library-manager-for-linux-macos-and-windows-vcpkg/).

**Comment by Ulfgard** [comment]
We happy take pull requests that update the CMakeLists.txt to incorporate vspkg package generation

**Comment by sursu** [comment]
Following the [Contribution Guidelines](https://github.com/Microsoft/vcpkg/blob/master/CONTRIBUTING.md) somebody will need to complete a Contributor License Agreement (CLA) before the pull request can be accepted.

**Comment by Ulfgard** [comment]
This is only needed for contributing to vcpkg(i.e. giving them your code). We are Shark, so no need to do that for us, except that you have to understand that everything we pull from you is going to be licensed under the LGPL as the rest of shark.

**Comment by Ulfgard** [comment]
Ah now i see. this is needed for adding the shark-vspkg into the official package list. We can do that once we know what to do and once the generation of those packages work.

**Comment by Ulfgard** [comment]
yeah, we can do that. But am i correct that it is already majorly hapeful if we can generate those packages our CMakeLists.txt so that you can install shark via package manager?

**Comment by shlublu** [comment]
Or a nuGet? I initially thought thiw was reserved to C#, but there are actually [plenty of C++ packages](https://www.nuget.org/packages?q=Tags%3A%22C%2B%2B%22) there.

**Comment by ras0219-msft** [comment]
Glad to see the interest in adding Shark to vcpkg!

We'd love to have this library in vcpkg; all you need to do is create an appropriate `ports/shark/` directory and make a PR! We enable users to download and build from source, so you don't need to generate any packages from your CMakeLists file. Anyone can add libraries, not just the authors :)

Some useful resources for creating the port are our zlib example[1] doc and existing ports for other cmake projects[2].

[1] https://github.com/Microsoft/vcpkg/blob/master/docs/examples/packaging-zlib.md
[2] https://github.com/Microsoft/vcpkg/blob/master/ports/cpprestsdk/portfile.cmake

**Comment by Ulfgard** [comment]
Thanks for the info. This looks easy enough. I will try to get a script working for the next release, which should only be a few weeks away.

**Comment by Ulfgard** [comment]
Good news. I managed to finish most of the script-part of integration. Shark compiled using MSVC 19  and i managed to use the package in a test program. I also used the opportunity to remove most warnings and a small number of recently introduced compile errors.

I am a bit worried about the "should run in visual studio 15" requirement I read in the vspkg doc. This might be...tough.

**Comment by ras0219-msft** [comment]
Don't worry about that too much; note that Visual Studio 15 actually means the latest release (Visual Studio 2017).
